test/integration/apiserver/apply/apply_test.go

```diff
@@ -267,6 +267,67 @@ func TestApplyUpdateApplyConflictForced(t *testing.T) {
 	}
 }
 
+// TestApplyGroupsManySeparateUpdates tests that when many different managers update the same object,
+// the number of managedFields entries will only grow to a certain size.
+func TestApplyGroupsManySeparateUpdates(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	obj := []byte(`{
+		"apiVersion": "admissionregistration.k8s.io/v1",
+		"kind": "ValidatingWebhookConfiguration",
+		"metadata": {
+			"name": "webhook",
+		},
+	}`)
+
+	object, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/admissionregistration.k8s.io/v1").
+		Resource("validatingwebhookconfigurations").
+		Name("webhook").
+		Param("fieldManager", "apply_test").
+		Body(obj).Do().Get()
+	if err != nil {
+		t.Fatalf("Failed to create object using Apply patch: %v", err)
+	}
+
+	for i := 0; i < 20; i++ {
+		time.Sleep(500 * time.Millisecond)
+		unique := fmt.Sprintf("updater%v", i)
+		path := "/apis/admissionregistration.k8s.io/v1"
+		if i%2 == 0 {
+			path = "/apis/admissionregistration.k8s.io/v1beta1"
+		}
+		object, err = client.CoreV1().RESTClient().Patch(types.MergePatchType).
+			AbsPath(path).
+			Resource("validatingwebhookconfigurations").
+			Name("webhook").
+			Param("fieldManager", unique).
+			Body([]byte(`{"metadata":{"labels":{"` + unique + `":"new"}}}`)).Do().Get()
+		if err != nil {
+			t.Fatalf("Failed to patch object: %v", err)
+		}
+	}
+
+	accessor, err := meta.Accessor(object)
+	if err != nil {
+		t.Fatalf("Failed to get meta accessor: %v", err)
+	}
+
+	actual, err := json.MarshalIndent(object, "\t", "\t")
+	if err != nil {
+		t.Fatalf("Failed to marshal object: %v", err)
+	}
+
+	// Expect 12 entries, because the cap for update entries is 10, and there are 2 apiVersions of the
+	// endpoint being updated, so there should also be two bucket entries.
+	if len(accessor.GetManagedFields()) > 12 {
+		t.Fatalf("Object contains more managers than expected:\n%v", string(actual))
+	}
```

Comment:
You could technically test that you get a conflict with the "manager-too-old-to-track" if you change any of the fields

Reply:
is this test still necessary with the unit tests I added? probably not